dotnet/src/Connectors/Connectors.Memory.Redis/RedisMemoryStore.cs

```diff
@@ -28,13 +28,65 @@ public sealed class RedisMemoryStore : IMemoryStore
+    /// <param name="vectorSize">Embedding vector size, defaults to 1536</param>
+    /// <param name="vectorIndexAlgorithm">Indexing algorithm for vectors, defaults to "HNSW"</param>
+    /// <param name="vectorType">Vector type, defaults to "FLOAT32"</param>
+    /// <param name="vectorDistanceMetric">Metric for measuring vector distances, defaults to "COSINE"</param>
+    /// <param name="queryDialect">Query dialect, must be 2 or greater for vector similarity searching, defaults to 2</param>
+    public RedisMemoryStore(IDatabase database, int vectorSize = 1536,
+        VectorIndexAlgorithms vectorIndexAlgorithm = VectorIndexAlgorithm,
+        VectorTypes vectorType = VectorType,
+        VectorDistanceMetrics vectorDistanceMetric = VectorDistanceMetric,
+        int queryDialect = QueryDialect)
     {
+        if (vectorSize <= 0)
+        {
+            throw new ArgumentException("Vector size must be a positive integer.");
+        }
+        if (!Enum.TryParse<Schema.VectorField.VectorAlgo>(vectorIndexAlgorithm.ToString(), out this._vectorIndexAlgorithm))
+        {
+            throw new ArgumentException("Unsupported vector indexing algorithm.");
+        }
+
         this._database = database;
         this._vectorSize = vectorSize;
         this._ft = database.FT();
+        this._vectorType = vectorType.ToString();
+        this._vectorDistanceMetric = vectorDistanceMetric.ToString();
+        this._queryDialect = queryDialect;
+    }
+
+    /// <summary>
+    /// Create a new instance of semantic memory using Redis.
+    /// </summary>
+    /// <param name="connectionString">Provide connection URL to a Redis instance</param>
+    /// <param name="vectorSize">Embedding vector size, defaults to 1536</param>
+    /// <param name="vectorIndexAlgorithm">Indexing algorithm for vectors, defaults to "HNSW"</param>
+    /// <param name="vectorType">Vector type, defaults to "FLOAT32"</param>
+    /// <param name="vectorDistanceMetric">Metric for measuring vector distances, defaults to "COSINE"</param>
+    /// <param name="queryDialect">Query dialect, must be 2 or greater for vector similarity searching, defaults to 2</param>
+    public RedisMemoryStore(string connectionString, int vectorSize = 1536,
+        VectorIndexAlgorithms vectorIndexAlgorithm = VectorIndexAlgorithm,
+        VectorTypes vectorType = VectorType,
+        VectorDistanceMetrics vectorDistanceMetric = VectorDistanceMetric,
+        int queryDialect = QueryDialect)
+    {
+        if (vectorSize <= 0)
+        {
+            throw new ArgumentException("Vector size must be a positive integer.");
+        }
+        if (!Enum.TryParse<Schema.VectorField.VectorAlgo>(vectorIndexAlgorithm.ToString(), out this._vectorIndexAlgorithm))
+        {
+            throw new ArgumentException("Unsupported vector indexing algorithm.");
+        }
+
+        this._database = ConnectionMultiplexer.Connect(connectionString).GetDatabase();
```

Comment:
I have a question. Is it necessary to consider managing and dispose `ConnectionMultiplexer` instances?

Reply:
Do you mean keeping a reference to the `ConnectionMultiplexer` within the memory store instance? I would think if someone has more complex use for the `ConnectionMultiplexer` that they should use the constructor that takes in an `IDatabase` instance so they can do their own management of it.

Reply:
@sneha-afk You are right. For complex scenarios when developers want to control disposable `ConnectionMultiplexer`, they can use constructor that accepts `IDatabase` interface. We use similar approach in AI Connectors for `HttpClient`. It's possible to pass custom `HttpClient` to control object lifetime, or we will provide our own instance of `HttpClient` as default option.

Reply:
In fact, I deliberately did not implement `IDisposable' at all, and did not allow the user to pass a connection string. The reason for this is mentioned in PR #1338, and I'm not sure which side of the issue this falls on. `IKernel` is registered as scoped lifetime with DI, `MemoryStore` is registered as singleton lifetime, and whenever `IKernel` is freed, it tries to dispose all internal instances that implement `IDisposable`. This results in the instances inside the MemoryStore being disposed of after the first request, unless the application is restarted.

https://github.com/microsoft/chat-copilot/blob/07f2fb09689aa007f22310db4769fe650d33af40/webapi/SemanticKernelExtensions.cs#L41

https://github.com/microsoft/chat-copilot/blob/main/webapi/SemanticKernelExtensions.cs#L99

https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/SemanticKernel/Kernel.cs#L250

https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/SemanticKernel/Memory/SemanticTextMemory.cs#L132